backend/app/core/security.py

```
@@ -424,7 +443,37 @@ def get_current_user_from_trusted_source(
-    # Get API trusted sources from configuration
+    # Import here to avoid circular imports
+    from app.models.api_key import APIKey, KEY_TYPE_SERVICE
+
+    # First, verify the service key in database
+    key_hash = hashlib.sha256(wegent_source.encode()).hexdigest()
+    service_key = (
+        db.query(APIKey)
+        .filter(
+            APIKey.key_hash == key_hash,
+            APIKey.key_type == KEY_TYPE_SERVICE,
+            APIKey.is_active == True,
```

Ruff (0.14.10)

Avoid equality comparisons to `True`; use `APIKey.is_active:` for truth checks

Replace with `APIKey.is_active`

(E712)